app/main.py

```
@@ -0,0 +1,327 @@
+            "count": len(tools),
+            "query": query
+        }
+        
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Error searching tools: {str(e)}")
+
+# Memory Management Endpoints
+@app.post("/memory/store")
+async def store_memory(
+    store_request: StoreRequest,
+    postgres_client: PostgresClient = Depends(get_postgres_client),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)